humilis_sam/resources.yaml.j2

```diff
@@ -1,11 +1,20 @@
 ---
 resources:
     {% if swagger_template %}
-    ApiGateway{{_layer.name}}{{_env.stage}}:
+    ApiKey:
+      Type: "AWS::ApiGateway::ApiKey"
+      DependsOn:
+        - ApiGateway
+      Properties:
+        Name: "{{_env.name}}-{{_env.stage|lower}}"
+        Description: "API key for stage {{_env.stage}} of environment {{_env.name}}"
+        Enabled: "true"
+
+    ApiGateway:
         Type: AWS::Serverless::Api
         Properties:
             DefinitionUri: {{swagger_template}}
-            StageName: {{_env.stage}}
+            StageName: {{_env.stage|lower}}
```

Comment:
I think it's better to be consistent and only use lowercase for elements that will be part of the URL endpoint of the API.